### PR TITLE
feat: fetch merchant key store only once per session

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Fetch merchant key store only once in session whenever Merchant Account is created. Merchant account fetch happens during authentication and we return MerchantAccount there. Added a seperate struct called `AuthenticationData` to fetch MerchantKeyStore with MerchantAccount and return. For the places where Merchant authentication is not used fetching will happen in place wherever needed.

## Motivation and Context

This removes redundant calls for merchant key store database.

## How did you test it?

Did payments using API Key and merchant key store only fetched once during authentication

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

**Comment by lsampras** [comment]
What does a session mean in this context?

Is it a payment_intent (& all it's api calls), or is it just one API request?

File: crates/common_utils/src/crypto.rs [hunk 1]
```diff
@@ -399,12 +399,23 @@ impl<T: Clone> Encryptable<T> {
     ///
     /// Get the inner data while consumping self
     ///
+    #[inline]
     pub fn into_inner(self) -> T {
         self.inner
     }
+
+    ///
+    /// Get the inner data while consumping self
```
Comment (Narayanbhat166) [inline crates/common_utils/src/crypto.rs:408]:
```suggestion
    /// Get the inner data while consuming self
```

File: crates/router/src/core/admin.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use common_utils::{
     ext_traits::ValueExt,
 };
 use error_stack::{report, FutureExt, ResultExt};
-use masking::Secret; //PeekInterface
+use masking::{PeekInterface, Secret}; //PeekInterface
```
Comment (SanchithHegde) [inline crates/router/src/core/admin.rs:8]:
```suggestion
use masking::{PeekInterface, Secret};
```

Remove this comment.

**Comment by dracarys18** [comment]
> What does a session mean in this context?
> 
> Is it a payment_intent (& all it's api calls), or is it just one API request?

It's one API request

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**